### ISSUE TITLE
Error when using VCL themes on a PromptForFileName dialog

### ISSUE THREAD

**Issue by AdminUser0** [RRUZ/vcl-styles-utils#212]
I am using themes (vsf) in my app. When I call  Vcl.Dialogs.PromptForFileName or MessageBox, FastMM detects that I try  to access an object that was already freed.    
1 FastMM shows an warning message (see FastMM report below)  
2 Then the debugger shows:      Debugger Exception Notification - Project WEditor.exe raised exception   class EOutOfMemory with message 'Out of memory'    
3 Finally the program continues normally and the PromptForFileName dialog opens normally (but it is not skinned). 

    function LoadSkinFromFile(CONST DiskShortName: string): Boolean;
    VAR  Style : TStyleInfo;
    begin
     Result:= FileExists(GetSkinDir+ DiskShortName);
    
     if Result then
      if TStyleManager.IsValidStyle(GetSkinDir+ DiskShortName, Style)
      then
        if NOT TStyleManager.TrySetStyle(Style.Name, FALSE)
        then
          begin
           TStyleManager.LoadFromFile(GetSkinDir+ DiskShortName);
           TStyleManager.SetStyle(Style.Name);
          end
        else Result:= FALSE
      else
         MesajError('Style is not valid: '+ GetSkinDir+ DiskShortName);
    end;

 The Fast MM report is:

    --------------------------------2017/8/28 15:05:18--------------------------------
    FastMM has detected an error during a GetMem operation. FastMM detected that a block has been modified after being freed. 
    
    Modified byte offsets (and lengths): 36(1)
    
    The previous block size was: 80
    
    This block was previously allocated by thread 0x45C, and the stack trace (return addresses) at the time was:
    406AC2 
    408027 
    40874E 
    63E2A6 [Vcl.SysStyles][Vcl][Vcl.SysStyles.TSysPopupStyleHook.Create]
    5DBF2A [Vcl.Themes][Vcl][Vcl.Themes.{System.Generics.Collections}TDictionary<System.string,Vcl.Themes.TSysStyleHookClass>.GetItem]
    5D799D [Vcl.Themes][Vcl][Vcl.Themes.AddControl]
    5D7ACB [Vcl.Themes][Vcl][Vcl.Themes.TCustomStyleEngine.HookCBProc]
    750A7459 [Unknown function at CallNextHookEx]
    750A9046 [Unknown function at SetWindowsHookExA]
    7749011A [KiUserCallbackDispatcher]
    750BDE30 [TrackPopupMenuEx]
    
    The block was previously used for an object of class: TSysPopupStyleHook
    
    The allocation number was: 103919
    
    The block was previously freed by thread 0x45C, and the stack trace (return addresses) at the time was:
    406ADE 
    408045 
    408799 
    63F377 [Vcl.SysStyles][Vcl][Vcl.SysStyles.TSysPopupStyleHook.Destroy]
    40814B 
    5D7B32 [Vcl.Themes][Vcl][Vcl.Themes.TCustomStyleEngine.HookCBProc]
    750A7459 [Unknown function at CallNextHookEx]
    750980B9 [Unknown function at GetUserObjectInformationW]
    75096DF8 [Unknown function at GetThreadDesktop]
    75098143 [Unknown function at GetUserObjectInformationW]
    7749011A [KiUserCallbackDispatcher]
    
    The current thread ID is 0x45C, and the stack trace (return addresses) leading to this error is:
    416169 [FastMM4][FastMM4][FastMM4.DebugGetMem]
    406AC2 
    409E23 
    40A054 
    5D7031 [Vcl.Themes][Vcl][Vcl.Themes.TStyleHook.GetText]
    65EB3B [Vcl.Styles][Vcl][Vcl.Styles.TSeStyleObject.FindObjectByAction]
    269336 [GetRawStackTrace]
    416354 [FastMM4][FastMM4][FastMM4.DebugFreeMem]
    416369 [FastMM4][FastMM4][FastMM4.DebugFreeMem]
    406ADE 
    408045 

Why do I get this error?

Notes:   
I am using [VCL Styles Utils] I just downloaded yesterday.
If I disable themes (load 'default win theme') the dialog works.

**Comment by RRUZ** [comment]
Please move the details from stackoverflow to here (it seems which your question will be closed soon). Also if you can't create a sample App there is not much help which I can give you. Please try to create a sample project to reproduce the issue.

**Comment by AdminUser0** [comment]
I had to move on from this because I need to finish something but I promise I will return and I will try to chop pieces from my program until I find the exact part that causes the problem to appear.

**Comment by AdminUser0** [comment]
I managed to reproduce it on demand:

1. Start the program. It will load the vsf skin
2. Click the 'Project' menu then the 'New' sub menu (which does nothing, just a dummy menu)
3.  Close the program
On shutdown Fast MM will detect a mem corruption in Vcl.SysStyles.TSysPopupStyleHook.Create. The the FastMM log provided. Voila :)




Here is the source code:

https://ufile.io/a7bwb
or
http://s000.tinyupload.com/?file_id=09126466747848393733

**Comment by RRUZ** [comment]
The project that you attached does not include any reference to this project. is only using the standard VCL Styles.

**Comment by AdminUser0** [comment]
oh... how do you include VCL Styles Utils? The documentation says that it is enough to have the path to the library included into the IDE Options.

**Comment by RRUZ** [comment]
That is the installation process of the library (just add the path to the IDE Options),   In order to use the library you must add the units reference to your project.  Anyway you are not using any unit of this project so the issue is related to the VCL Styles and not to this project. 

  Btw I tried your demo App in RAD Studio 10.2 and not exception is shown, So maybe this issue was fixed on the recent versions of Delphi.
